sklearn/preprocessing/_encoders.py

```diff
@@ -651,18 +783,31 @@ def inverse_transform(self, X):
         found_unknown = {}
 
         for i in range(n_features):
-            n_categories = len(self.categories_[i])
+            if self.drop is None or self.drop_idx_[i] is None:
+                cats = self.categories_[i]
+            else:
+                cats = np.delete(self.categories_[i], self.drop_idx_[i])
+            n_categories = len(cats)
+            if n_categories == 0:
+                X_tr[:, i] = self.categories_[i][self.drop_idx_[i]]
+                j += n_categories
+                continue
             sub = X[:, j:j + n_categories]
-
             # for sparse X argmax returns 2D matrix, ensure 1D array
             labels = np.asarray(_argmax(sub, axis=1)).flatten()
-            X_tr[:, i] = self.categories_[i][labels]
-
-            if self.handle_unknown == 'ignore':
-                # ignored unknown categories: we have a row of all zero's
+            X_tr[:, i] = cats[labels]
+            if self.handle_unknown == 'ignore' or self.drop is not None:
                 unknown = np.asarray(sub.sum(axis=1) == 0).flatten()
                 if unknown.any():
-                    found_unknown[i] = unknown
+                    # In this case, we can safely assume that all of the nulls
+                    # in each column are the dropped value
+                    if self.drop is not None:
+                        X_tr[unknown, i] = (self.categories_[i][
+                                                self.drop_idx_[i]])
```

Comment:
if you re-use the already defined `cats` (instead of `self.categories_[i]`) it might fit on a single line